Lib/queue.py

```diff
@@ -198,6 +226,27 @@ def get_nowait(self):
         '''
         return self.get(block=False)
 
+    def shutdown(self, immediate=False):
+        '''Shut-down the queue, making queue gets and puts raise.
+
+        By default, gets will only raise once the queue is empty. Set
+        'immediate' to True to make gets raise immediately instead.
+
+        All blocked callers of put() will be unblocked, and also get()
+        and join() if 'immediate'. The ShutDown exception is raised.
+        '''
+        with self.mutex:
+            self.is_shutdown = True
+            if immediate:
+                while self._qsize():
+                    self._get()
```

Comment:
Why not `self.queue.clear()`?

Reply:
I don't believe it's worth potentially breaking user subclasses which reimplement `_qsize` and `_get` for a minor performance gain, even if it's not the documented API